plugins/Files/js/components/renamedialog.js

```diff
@@ -0,0 +1,34 @@
+import React, { PropTypes } from 'react'
+import Path from 'path'
+
+const RenameDialog = ({siapath, actions}) => {
+	const onYesClick = (e) => {
+		e.preventDefault()
+		actions.renameFile(siapath, Path.join(Path.dirname(siapath), e.target.newsiapath.value))
```

Comment:
since the form value is the basename of the file, not the full path, I think it makes sense to change the input's name to `newname` or something similar.